crossbar/router/broker.py

```diff
@@ -171,13 +171,17 @@ def _filter_publish_receivers(self, receivers, publish):
 
         # if "eligible_authid" we only accept receivers that have the correct authid
         if publish.eligible_authid:
+            eligible = set()
             for aid in publish.eligible_authid:
-                receivers = receivers & self._router._authid_to_sessions.get(aid, set())
```

Comment:
I believe this is not correct. You can not iteratively intersect with the sessions of the elements of the eligible_authid list. something like this should work I believe
```python
        if publish.eligible_authid:
            eligible = set()
            for aid in publish.eligible_authid:
                eligible = eligible | self._router._authid_to_sessions.get(aid, set())
            receivers = receivers & eligible
```

(same below for authrole)